src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.cpp

```diff
@@ -1243,131 +1243,132 @@ class npc_announcer_toc5 : public CreatureScript
+                {
+                    // If Grand Champions encounter hasn't been started and the player is not mounted
+                    if (player->GetTeam() == HORDE)
+                        SendGossipMenuFor(player, GOSSIP_TEXT_UNMOUNTED_H, me->GetGUID());
+                    else
+                        SendGossipMenuFor(player, GOSSIP_TEXT_UNMOUNTED_A, me->GetGUID());
+                }
+                else if (instance->GetBossState(DATA_GRAND_CHAMPIONS) == DONE && (instance->GetBossState(DATA_ARGENT_CHALLENGE) == NOT_STARTED || instance->GetBossState(DATA_ARGENT_CHALLENGE) == TO_BE_DECIDED))
+                {
+                    // If Grand Champions encounter is done and Eadric the Pure nor Argent Confessor Paletress encounters have been started
+                    AddGossipItemFor(player, GOSSIP_ICON_CHAT, GOSSIP_START_EVENT2, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 1);
+                    SendGossipMenuFor(player, GOSSIP_TEXT_SECOND_BOSS, me->GetGUID());
+                }
+                else if (instance->GetBossState(DATA_ARGENT_CHALLENGE) == DONE && (instance->GetBossState(DATA_BLACK_KNIGHT) == NOT_STARTED || instance->GetBossState(DATA_BLACK_KNIGHT) == TO_BE_DECIDED))
+                {
+                    // If Grand Champions, Eadric the Pure and Argent Confessor Paletress encounters are all done but Black Knight encounter has not been started
+                    AddGossipItemFor(player, GOSSIP_ICON_CHAT, GOSSIP_START_EVENT1, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 1);
+                    SendGossipMenuFor(player, GOSSIP_TEXT_THIRD_BOSS, creature->GetGUID());
+                }
+            }
             return true;
+        }
 
-        bool seen = true;
-        Map::PlayerList const& players = player->GetMap()->GetPlayers();
-        for (Map::PlayerList::const_iterator itr = players.begin(); itr != players.end(); ++itr)
+        bool GossipSelect(Player* player, uint32 /*menuId*/, uint32 gossipListId) override
         {
-            if (Player const *plr = itr->GetSource())
+            uint32 const action = player->PlayerTalkClass->GetGossipOptionAction(gossipListId);
+            ClearGossipMenuFor(player);
+            CloseGossipMenuFor(player);
+            switch (action)
             {
-                // if everyone from your group have completed one of the Trial of the Champion achievements, you have option to skip the event
-                // maybe not the correct way to do it but I couldn't figure out better
-                if (!plr->HasAchieved(4298) /* Heroic ToC (alliance) */ && !plr->HasAchieved(3778) /* Normal ToC (horde) */ && !plr->HasAchieved(4297) /* Heroic ToC (horde) */ && !plr->HasAchieved(4296) /* Normal ToC (alliance) */)
-                {
-                    seen = false;
+                case GOSSIP_ACTION_INFO_DEF + 1:
+                    ENSURE_AI(npc_announcer_toc5::npc_announcer_toc5AI, me->AI())->StartEncounter();
```

Comment:
these casts seem useless

Reply:
Well, I'm fixing compile errors, all these casts for you 😘

Reply:
that is part of fixing compile errors

Reply:
Now  travis  seems happy....
And I don't know very  well how change these casts xD

Reply:
just remove `ENSURE_AI(npc_announcer_toc5::npc_announcer_toc5AI, me->AI())->`

Reply:
OK, so it should work with only `StartEncounter();` on this line? 😅

Reply:
yes

Reply:
Now it's ok @joschiwald ??